fix #175031: selectable palette in Note Group UI.

ericfont opened the pull request:
Clicking notes toggles between palette's currently selected beam mode and auto.
Initial default selected beam mode is BEGIN, which will keep default behavior similar behavior prior to this commit.

ericfont commented on mscore/noteGroups.cpp:
```diff
@@ -86,6 +86,8 @@ NoteGroups::NoteGroups(QWidget* parent)
       populateIconPalette(iconPalette, bpa);
       iconPalette->setReadOnly(true);
+      iconPalette->setSelectable(true);
+      iconPalette->setSelected(0);
```
This sets initial selected state of palette to Beam::Mode::BEGIN.

ericfont commented on mscore/noteGroups.cpp:
```diff
@@ -139,6 +141,8 @@ Groups NoteGroups::groups()
       {
       setSig(_sig, _groups);
+      iconPalette->setSelected(0);
+      iconPalette->update();
```
have to force update here to trigger redraw of palette reflect resetting selecting beam mode back to the default.

ericfont commented on mscore/noteGroups.cpp:
```diff
@@ -148,10 +152,31 @@ void NoteGroups::resetClicked()
+            chord->setBeamMode(selectedBeamMode);
+      else
             chord->setBeamMode(Beam::Mode::AUTO);
+
```
note that behavior of above if/else is slightly different than was previously, but I much prefer my new behavior.  With the old version, clicking on a note that wasn't either AUTO or BEGIN would do nothing, so I'm afraid user would be confused and think that the UI was simply not working.  With my new behavior, clicking on a note will always do something to the beam mode.

lasconic commented:
See https://musescore.org/fr/node/175031#comment-660186
It also needs to be rebased if we wanted to merge it.

ericfont commented:
Rebased and made a few changes: 
- Increased the clickable region such that click event will change beam mode of nearest note (so can click on whitespace).
- Clicking notes toggles between palette's currently selected beam mode and auto.
- Initial default keeps prior behavior whereby togging works the same (I'm using the same if statement)
- If an palette cell is selected, then click on notes will change to the selected beam property.  (So no longer will toggle if a palette cell is selected).
- Reset button will now remove the palette's selection.
- Removed the unnecessary labels "1/8", "1/16", "1/32" from the UI.
- Moved the hspacer on the UI to be to the right of the palette.

dmitrio95 commented:
Needs actualization for current master.
